core/stopWatch.go

```diff
@@ -76,8 +77,35 @@ func (sw *stopWatch) GetMeasurements() []interface{} {
 	return output
 }
 
+// GetMeasurementsMap returns the measurements as a map of (identifier, duration in seconds)
+func (sw *StopWatch) GetMeasurementsMap() map[string]float64 {
```

Comment:
`float64`, `float32`, `time.Duration`, `int` (nanoseconds) were all candidates :) 

 - `time.Duration` - semantic primitive, more detailed, better semantics, but client code has to look to its inner fields in **practice, most of the time**, they should have better knowledge about its fields (law of Demeter)
 - `float` - raw primitive, less semantics, easier client code. 

So I chose `float`. Should we change?

Reply:
time.Duration does a pretty print when `String` method is called. We have a lot of time.Duration uses in our project and since this struct uses time.Duration anyway, I suggest we should use time.Duration.